image/image.go

```diff
@@ -310,11 +311,20 @@ func bootstrapToRootDir(tsto *ToolingStore, model *asserts.Model, opts *Options,
 		}
 	}
 
+	baseName := defaultCore
+	if model.Base() != "" {
+		baseName = model.Base()
+	}
+
 	snaps := []string{}
-	// core,kernel,gadget first
-	snaps = append(snaps, local.PreferLocal(defaultCore))
+	// core/base,kernel,gadget first
+	snaps = append(snaps, local.PreferLocal(baseName))
 	snaps = append(snaps, local.PreferLocal(model.Kernel()))
 	snaps = append(snaps, local.PreferLocal(model.Gadget()))
+	// always add an implicit snapd when a base is used
+	if model.Base() != "" {
+		snaps = append(snaps, "snapd")
```

Comment:
it needs to be the first one I think or there will be problems with hooks of gadget etc